### ISSUE TITLE
Add timestamps to tag_selection and node_tags tables

### ISSUE THREAD

**Issue by jywarren** [publiclab/plots2#5793]
We need a new migration to add `created_at` and `updated_at` to these two tables; this will help solve https://github.com/publiclab/plots2/issues/4925, but also will help us add ordering to our pinned nodes (https://github.com/publiclab/plots2/issues/766), so that they are pinned in order -- first pinned is first displayed:

https://github.com/publiclab/plots2/blob/35e10a2eeabdf5afce7bb67704548435997dc58a/app/models/concerns/node_shared.rb#L415-L419

We'd love help with this!

> Active Record automatically timestamps create and update operations if the table has fields named created_at/created_on or updated_at/updated_on.

**Comment by siaw23** [comment]
What are the two tables?

**Comment by jywarren** [comment]
I believe it's tag_selection and term_node!

**Comment by skilfullycurled** [comment]
@jywarren, this is great, will this actually be a return to the previous way that tags were entered in the database?  At least prior to 2016, tags did have timestamps for the first time they were created, and I believe a time stamp for when they were added to a node.  I'll have to look to see if this would change anything to the tags download feature which right now downloads all of the term_nodes because they do not have timestamps anymore.

**Comment by jywarren** [comment]
Huh - i wonder how that got changed -- i think we simplified the tables but
perhaps didn't preserve these features when we should've! But yes, this
would be super.

**Comment by skilfullycurled** [comment]
Yeah, there used to be a table called "tags" which was [removed in a commit](https://github.com/publiclab/plots2/commit/eff8a442a6b51a7d00fc462d5e039c82e1b4ea52#diff-4ef680bc0c8c6ec87925705fae6c0a1cL360) October 8, 2016 (I think).

**Comment by skilfullycurled** [comment]
@jywarren, I've been researching migrations to see if this was something I could help on.

It seems like there is an internet consensus (n=3) that in Rails 5+ adding created_at and updated_at is done in the following [manner](https://stackoverflow.com/a/46521142/6023530) (also see [this](https://medium.com/@ebrugulec/add-timestamps-to-existing-table-in-db-rails-5-9e185cfab37e) and [this](https://makandracards.com/makandra/19325-your-database-tables-should-always-have-timestamps) which are slightly different but I think less clear versions of the same). 

```
class AddTimestampsToTags < ActiveRecord::Migration[5.2]
  def change
    # add new column but allow null values
    add_timestamps :term_node, null: true 
    add_timestamps :tag_selections, null: true 

    # backfill existing record with created_at and updated_at
    # values making clear that the records are faked
    long_ago = DateTime.new(2000, 1, 1)
    SeeQuestionTwoBelow.update_all(created_at: long_ago, updated_at: long_ago)
    TagSelection.update_all(created_at: long_ago, updated_at: long_ago)
    
    # change not null constraints
    change_column_null :term_node, :created_at, false
    change_column_null :tag_selections, :updated_at, false

  end
end
```
1) Is this the correct strategy?  If so...

2) The tables in question are term_node and tag_selections.  The tag_selection.rb contains the class TagSelection but I'm not sure of the class for the term_node table since it does not have an obvious model. There is a Tag model whose table is term_data, and a NodeTag whose table is community_tags.  Is it either of those?

3) The back fill seems to be required else you can't have null be false but for our purposes it seems like it would be useful to try and retrieve them which I think we can by referencing other tables.  Is that worth doing, or should we just use the backfill strategy above?

4) After a migration is created, will it add the changes to the model or need the models be changed as well.

**Comment by cesswairimu** [comment]
Thanks @skilfullycurled, for pointing the other table for me. @jywarren could you please explain a little for me how/where we use `term_node` table  so I can see how I can integrate it in getting subscriptions stats thanks

**Comment by skilfullycurled** [comment]
@jywarren, can we close this since it was included in PR #5875 which fixed #4925?

**Comment by jywarren** [comment]
Yes, thanks so much!
